### ISSUE TITLE
unnecessary_lambda_linter() gets confused by indexing in lambda function

### ISSUE THREAD

**Issue by Bisaloo** [r-lib/lintr#2231]
``` r
lintr::lint(
  text = "lapply(l, function(x) rle(x)$values)",
  linters = lintr::unnecessary_lambda_linter()
)
#> <text>:1:11: warning: [unnecessary_lambda_linter] Pass rle directly as a symbol to lapply() instead of wrapping it in an unnecessary anonymous function. For example, prefer lapply(DF, sum) to lapply(DF, function(x) sum(x)).
#> lapply(l, function(x) rle(x)$values)
#>           ^~~~~~~~~~~~~~~~~~~~~~~~

lintr::lint(
  text = "lapply(l, function(x) rle(x)[['values']])",
  linters = lintr::unnecessary_lambda_linter()
)
#> <text>:1:11: warning: [unnecessary_lambda_linter] Pass rle directly as a symbol to lapply() instead of wrapping it in an unnecessary anonymous function. For example, prefer lapply(DF, sum) to lapply(DF, function(x) sum(x)).
#> lapply(l, function(x) rle(x)[['values']])
#>           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<sup>Created on 2023-10-06 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I believe these are valid cases and shouldn't create lints

**Comment by MichaelChirico** [comment]
nice find, thanks
